src/transfer.ts

```diff
@@ -25,6 +25,45 @@ export const transferStatusSchema = z.nativeEnum(TransferStatus, {
   description: 'transferStatusSchema',
 })
 
+// These enums are first segmented between transfers in and transfers out
+// since they both allow a slightly different set of statuses.
+// Transfer in statuses are further segmented, since once a transfer in
+// progresses to sending the user funds, providers are expected
+// to return new metadata about the transfer. This segmentation allows
+// us to encode those semantics into Zod schemas.
+export enum TransferInStatusPreTx {
```

Comment:
I had originally wanted to "merge" native TS enums here; that is, create `TransferInStatusPreTx` and `TransferInStatusPostTx`, and have `TransferInStatus` be a new enum created by "merging" the first two. TLDR, this isn't possible in TS while playing nicely with Zod's `nativeEnum`, so I opted for a different approach here by re-using enum values so as not to duplicate string literals.

Reply:
i had a different idea for this that may interest you, commented on it below